airbyte-ci/connectors/pipelines/pipelines/tests/python_connectors.py

```diff
@@ -55,62 +54,116 @@ async def _run(self) -> StepResult:
+            config_file_name = self.PYTEST_INI_FILE_NAME
+            test_config = await self.context.get_connector_dir(include=[self.PYTEST_INI_FILE_NAME]).file(self.PYTEST_INI_FILE_NAME)
+            self.logger.info(f"Found {self.PYTEST_INI_FILE_NAME}, using it for testing.")
+        elif self.PYPROJECT_FILE_NAME in connector_dir_entries:
+            config_file_name = self.PYPROJECT_FILE_NAME
+            test_config = await self.context.get_connector_dir(include=[self.PYTEST_INI_FILE_NAME]).file(self.PYTEST_INI_FILE_NAME)
+            self.logger.info(f"Found {self.PYPROJECT_FILE_NAME} at connector level, using it for testing.")
+        else:
+            config_file_name = f"global_{self.PYPROJECT_FILE_NAME}"
+            test_config = await self.context.get_repo_dir(include=[self.PYPROJECT_FILE_NAME]).file(self.PYPROJECT_FILE_NAME)
+            self.logger.info(f"Found {self.PYPROJECT_FILE_NAME} at repo level, using it for testing.")
+
+        def prepare_for_testing(built_connector_container: Container) -> Container:
+            return (
+                built_connector_container
+                # Reset the entrypoint
+                .with_entrypoint([])
+                # Mount the connector directory in /test_environment
+                # For build optimization the full directory is not mounted by default
+                # We need the setup.py/pyproject.toml and the tests code to be available
+                # Install the extra dependencies
+                .with_mounted_directory("/test_environment", connector_dir)
+                # Jump in the /test_environment directory
+                .with_workdir("/test_environment").with_mounted_file(config_file_name, test_config)
+                # Mount the secrets
+                .with_(secret_mounting_function)
+                # Install the extra dependencies
+                .with_exec(["pip", "install", f".[{','.join(extra_dependencies_names)}]"], skip_entrypoint=True)
+                # Execute pytest on the test directory
+                .with_exec(
+                    [
+                        "python",
+                        "-m",
+                        "pytest",
+                        "-s",
+                        self.test_directory_name,
+                        "-c",
+                        config_file_name,
+                    ]
+                )
+            )
+
+        return prepare_for_testing
 
-        connector_under_test = connector_under_test.with_(environments.bound_docker_host(self.context)).with_(
-            await environments.mounted_connector_secrets(self.context, "secrets")
-        )
-        return await self._run_tests_in_directory(connector_under_test, "integration_tests")
+
+class UnitTests(PytestStep):
+    """A step to run the connector unit tests with Pytest."""
+
+    title = "Unit tests"
+    test_directory_name = "unit_tests"
+
+
+class IntegrationTests(PytestStep):
```

Comment:
I love that these got simpler!

I do often try to get us away from classes and inheritance.

But this is a good fit.

Dropping a link to [Liskov Substitution Principle (LSP) ](https://stackoverflow.com/questions/56860/what-is-an-example-of-the-liskov-substitution-principle) for reference :)